### PR TITLE
[YAML] Fix one-liners to work with multiple new lines

### PR DESCRIPTION
**Pull request by alexpott**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | none |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/Yaml/Parser.php [hunk 1]
```diff
@@ -234,9 +234,11 @@ public function parse($value, $exceptionOnInvalidType = false, $objectSupport =
...
                 }
 
-                // 1-liner optionally followed by newline
-                $lineCount = count($this->lines);
-                if (1 === $lineCount || (2 === $lineCount && empty($this->lines[1]))) {
+                // store the last PCRE regex error code
+                $pregLastError = preg_last_error();
+
+                // 1-liner optionally followed by newline(s)
+                if (preg_match('#^.+\s*#i', $value)) {
```
Comment (fabpot) [inline src/Symfony/Component/Yaml/Parser.php]: what about trimming instead? `trim($value)`

Comment (alexpott) [inline src/Symfony/Component/Yaml/Parser.php]: I tried to do that in Parser::cleanup() but that broke things... but doing it here might work

Comment (alexpott) [inline src/Symfony/Component/Yaml/Parser.php]: Nope I get failures:
1) Symfony\Component\Yaml\Tests\ParserTest::testEndOfTheDocumentMarker
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'foo'
+'--- %YAML:1.0
+foo
+...'

2) Symfony\Component\Yaml\Tests\ParserTest::testUnindentedCollectionException
Failed asserting that exception of type "\Symfony\Component\Yaml\Exception\ParseException" is thrown.

**Comment by fabpot** [comment]
Thank you @alexpott.

**Pull request closed without merging**